Broadcastify Calls API Prints Stacktrace on Socket Reset Exception

Update the request to catch/handle a CompletionException that has an embedded SocketException.


'''
10:08:51.519 ERROR i.g.d.a.b.b.BroadcastifyCallBroadcaster - Error while sending upload URL request  [469MB/1GB 32%]
java.util.concurrent.CompletionException: java.net.SocketException: Connection reset
        at java.base/java.util.concurrent.CompletableFuture.encodeRelay(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.completeRelay(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
        at java.net.http/jdk.internal.net.http.Stream.completeResponseExceptionally(Unknown Source)
        at java.net.http/jdk.internal.net.http.Stream.cancelImpl(Unknown Source)
        at java.net.http/jdk.internal.net.http.Stream.connectionClosing(Unknown Source)
        at java.net.http/jdk.internal.net.http.Http2Connection.shutdown(Unknown Source)
        at java.net.http/jdk.internal.net.http.Http2Connection$Http2TubeSubscriber.processQueue(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SynchronizedRestartableTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.net.SocketException: Connection reset
        at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(Unknown Source)
        at java.base/sun.nio.ch.SocketChannelImpl.read(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube.readAvailable(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$InternalReadSubscription.read(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$SocketFlowTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$InternalReadSubscription.signalReadable(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$ReadEvent.signalEvent(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$SocketFlowEvent.handle(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(Unknown Source)
        at java.base/java.util.ArrayList.forEach(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(Unknown Source)
10:16:11.404 ERROR i.g.d.a.b.b.BroadcastifyCallBroadcaster - Error while sending upload URL request  [1GB/1GB 72%]
java.util.concurrent.CompletionException: java.net.SocketException: Connection reset
        at java.base/java.util.concurrent.CompletableFuture.encodeRelay(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.completeRelay(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
        at java.net.http/jdk.internal.net.http.Stream.completeResponseExceptionally(Unknown Source)
        at java.net.http/jdk.internal.net.http.Stream.cancelImpl(Unknown Source)
        at java.net.http/jdk.internal.net.http.Stream.connectionClosing(Unknown Source)
        at java.net.http/jdk.internal.net.http.Http2Connection.shutdown(Unknown Source)
        at java.net.http/jdk.internal.net.http.Http2Connection$Http2TubeSubscriber.processQueue(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SynchronizedRestartableTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.net.SocketException: Connection reset
        at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(Unknown Source)
        at java.base/sun.nio.ch.SocketChannelImpl.read(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube.readAvailable(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$InternalReadSubscription.read(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$SocketFlowTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(Unknown Source)
        at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$InternalReadSubscription.signalReadable(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$ReadEvent.signalEvent(Unknown Source)
        at java.net.http/jdk.internal.net.http.SocketTube$SocketFlowEvent.handle(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(Unknown Source)
        at java.base/java.util.ArrayList.forEach(Unknown Source)
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(Unknown Source)
'''

--- a/src/main/java/io/github/dsheirer/audio/broadcast/broadcastify/BroadcastifyCallBroadcaster.java
+++ b/src/main/java/io/github/dsheirer/audio/broadcast/broadcastify/BroadcastifyCallBroadcaster.java
@@ -48,6 +48,7 @@ import java.net.http.HttpResponse;
 import java.time.Duration;
 import java.util.List;
 import java.util.Queue;
+import java.util.concurrent.CompletionException;
 import java.util.concurrent.LinkedTransferQueue;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
@@ -240,14 +241,14 @@ public class BroadcastifyCallBroadcaster extends AbstractAudioBroadcaster<Broadc
                         .whenComplete((stringHttpResponse, throwable) -> {
                             if(throwable != null || stringHttpResponse.statusCode() != 200)
                             {
-                                if(throwable instanceof IOException)
+                                if(throwable instanceof IOException || throwable instanceof CompletionException)
                                 {
                                     //We get socket reset exceptions occasionally when the remote server doesn't
                                     //fully read our request and immediately responds.
                                 }
                                 else
                                 {
-                                    mLog.error("Error while sending upload URL request", throwable);
+                                    mLog.error("Error while sending upload URL request" + throwable.getLocalizedMessage());
                                     setBroadcastState(BroadcastState.TEMPORARY_BROADCAST_ERROR);
                                 }
                                 incrementErrorAudioCount();
@@ -284,7 +285,7 @@ public class BroadcastifyCallBroadcaster extends AbstractAudioBroadcaster<Broadc
                                             .whenComplete((fileResponse, throwable1) -> {
                                                 if(throwable1 != null || fileResponse.statusCode() != 200)
                                                 {
-                                                    if(throwable1 instanceof IOException)
+                                                    if(throwable1 instanceof IOException || throwable1 instanceof CompletionException)
                                                     {
                                                         //We get socket reset exceptions occasionally when the remote server doesn't
                                                         //fully read our request and immediately responds.

